editor/components/block-list/block.js

```diff
@@ -428,6 +434,43 @@ export class BlockListBlock extends Component {
 		this.setState( { error } );
 	}
 
+	onDragStart( event ) {
+		this.props.onDragStart(
+			event,
+			this.props.uid,
+			`block-${ this.props.uid }`,
+			this.props.order,
+			BLOCK_REORDER
+		);
+		// Following fires after the events in props.onDragStart.
+		setTimeout( () => {
```

Comment:
Are the timeouts necessary here?

Reply:
>Are the timeouts necessary here?

 Especially because we have no guarantee the component is still mounted after the scheduled timeout occurs.

Reply:
Might be obvious at this point, but `setTimeout` is a [code smell](https://en.wikipedia.org/wiki/Code_smell), often signals a misunderstanding of code stack execution flows, and makes maintenance and future debugging very difficult.

Reply:
Agreed. I need to investigate these couple of timeouts here as they were added at the very end after moving the drag logic to an external component/HOC. Please understand this has gone through quite a few iterations, and especially since adding the `withDragging` HOC some things started to fall apart. I will have more on this on Saturday, hopefully we can get rid of them without an issue.